torii/src/lib.rs

```diff
@@ -85,177 +94,182 @@ impl Torii {
-                    if let Err(error) =
-                        routing::subscription::handle_subscription(events, this_ws).await
-                    {
-                        iroha_logger::error!(%error, "Failure during subscription");
+            .route(
+                uri::QUERY,
+                post({
+                    let query_service = self.query_service.clone();
+                    let state = self.state.clone();
+                    move |ClientQueryRequestExtractor(query_request): ClientQueryRequestExtractor| {
+                        routing::handle_queries(query_service, state, query_request)
                     }
-                })
-            });
-
-        // `warp` panics if there is `/` in the string given to the `warp::path` filter
-        // Path filter has to be boxed to have a single uniform type during iteration
-        let block_ws_router_path = uri::BLOCKS_STREAM
-            .split('/')
-            .skip_while(|p| p.is_empty())
-            .fold(warp::any().boxed(), |path_filter, path| {
-                path_filter.and(warp::path(path)).boxed()
-            });
+                }),
+            )
+            .route(
+                uri::CONFIGURATION,
+                post({
+                    let kiso = self.kiso.clone();
+                    move |Json(config): Json<_>| routing::handle_post_configuration(kiso, config)
+                }),
+            );
 
-        let blocks_ws_router = block_ws_router_path
-            .and(add_state!(self.kura))
-            .and(warp::ws())
-            .map(|sumeragi: Arc<_>, ws: Ws| {
-                ws.on_upgrade(|this_ws| async move {
-                    if let Err(error) = routing::handle_blocks_stream(sumeragi, this_ws).await {
-                        iroha_logger::error!(%error, "Failed to subscribe to blocks stream");
+        let router = router
+            .route(uri::SUBSCRIPTION, get({
+                let events = self.events.clone();
+                move |ws: WebSocketUpgrade| {
+                    core::future::ready(
+                        ws.on_upgrade(|ws| async move {
+                            if let Err(error) =
+                                routing::subscription::handle_subscription(events, ws).await
+                            {
+                                iroha_logger::error!(%error, "Failure during subscription");
+                            }
+                        })
+                    )
+                }
+            }))
+            .route(uri::BLOCKS_STREAM,
+                post({
+                    let kura = self.kura.clone();
+                    move |ws: WebSocketUpgrade| {
+                        core::future::ready(
+                            ws.on_upgrade(|ws| async move {
+                                if let Err(error) = routing::handle_blocks_stream(kura, ws).await {
+                                    iroha_logger::error!(%error, "Failed to subscribe to blocks stream");
```

Comment:
nit: This error can be logged not only due to "failure to subscribe", but also due to some error happening when handling the subscription (the future lives as long as the client is interested in the stream).